control-plane/subcommand/mesh-init/command.go

```diff
@@ -170,6 +174,13 @@ func (c *Command) Run(args []string) int {
 		}
 	}
 
+	// Write the proxy ID to the shared volume so `consul connect envoy` can use it for bootstrapping.
+	if err = common.WriteFileWithPerms(c.flagProxyIDFile, c.flagProxyName, 0444); err != nil {
```

Comment:
Where does this file end up being consumed? I can't find anywhere that sidecars depend on having this file present, so I'm surprised that mesh gateways would

Reply:
We talked a bit about this elsewhere and are thinking that the `../proxyid` file is probably not required for v2, even though it was for v1.